lamindb/_curate.py

```diff
@@ -1168,8 +1171,12 @@ def _check_save_keys(self):
         with _open_tiledbsoma(self._experiment_uri, mode="r") as experiment:
             experiment_obs = experiment.obs
             self._n_obs = len(experiment_obs)
-            valid_obs_keys = [k for k in experiment_obs.keys() if k != "soma_joinid"]
+            valid_obs_keys = [
+                k for k in experiment_obs.schema.names if k != "soma_joinid"
```

Comment:
Please read from the saved schema here to avoid redundant requests, maybe they cache but let’s be careful anyways.

Reply:
You save the schema on the next line anyways.

Reply:
Done!